sklearn/cluster/_kmeans.py

```diff
@@ -1057,7 +1057,9 @@ def fit(self, X, y=None, sample_weight=None):
                 x_squared_norms=x_squared_norms, n_threads=self._n_threads)
 
             # determine if these results are the best so far
-            if best_inertia is None or inertia < best_inertia:
+            # allow small tolerance on the inertia to accomodate for rounding
+            # errors.
```

Comment:
maybe mention that rounding errors are not deterministic because of OpenMP parallelism.